### PR TITLE
test(module:badge): added missing tests

### PR DESCRIPTION
**Pull request by ParsaArvanehPA**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [✔] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [✔] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Application (the showcase website) / infrastructure changes
- [✔] Other... Please describe: missing unit tests for the Badge Component

## What is the new behavior?
|               | Before  | After  |
|---------------|---------|--------|
| **Statements**| 89.95%  | 100%   |
| **Branches**  | 64.28%   | 100%   |
| **Functions** | 85.71%  | 100%   |
| **Lines**     | 89.7%  | 100%   |

## Does this PR introduce a breaking change?
- [ ] Yes
- [✔] No

### CODE REVIEW

File: components/badge/badge.spec.ts [hunk 1]
```diff
@@ -237,3 +239,117 @@ export class NzTestBadgeRtlComponent {
   direction: Direction = 'rtl';
   count = 5;
 }
+
+describe('badge', () => {
+  let fixture: ComponentFixture<NzBadgeComponent>;
+  let component: NzBadgeComponent;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
```
Comment (Laffery) [inline components/badge/badge.spec.ts:248]: `NzBadgeComponent` is a standalone component, no need to import here, use `TestBed.createComponent()` directly

File: components/badge/badge.spec.ts [hunk 2]
```diff
@@ -237,3 +239,117 @@ export class NzTestBadgeRtlComponent {
...
+      }
+    };
+    component.nzColor = color;
+    component.ngOnChanges(change);
+    tick();
+    expect(component.presetColor).toEqual(null);
+  }));
+});
+
+describe('ribbon', () => {
```
Comment (Laffery) [inline components/badge/badge.spec.ts:294]: suggested to be nested in the `badge` suite

**Review by Laffery — CHANGES_REQUESTED** [review]

**Comment by ParsaArvanehPA** [comment]
@Laffery 
Thank you for reviewing this PR; I have made the requested changes.

**Review by Laffery — APPROVED** [review]
LGTM, waiting for the CI passed

**Pull request merged**